### PR TITLE
LF-3332: Fix npm vulnerable packages

### PR DESCRIPTION
**Pull request by SayakaOno**
**Description**

### webapp
- fix packages by running `pnpm audit --fix`
- uninstall `xlsx`

- there is [one remaining issue](https://github.com/advisories/GHSA-353f-5xf4-qw67), but our app doesn't seem to be affected.
  > Only users explicitly exposing the Vite dev server to the network (using --host or [server.host config option](https://vitejs.dev/config/server-options.html#server-host)) are affected, and only files in the immediate Vite project root folder could be exposed.

### api
- `npm audit fix`
- update `jwks-rsa`
- update `express-jwt`. (most of the changes in this PR are related to this) README: [Migration from v6](https://github.com/auth0/express-jwt#migration-from-v6)
  - `req.user` -> `req.auth`
  - `import jwt from ...` -> `import { expressjwt } from ...`
- [override](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#overrides) `glob-parent`. There is a [bug](https://github.com/npm/cli/issues/5443) that `overrides` does not work after running `npm install`. `npm update` needs to be run to make the `overrides` work.

- there are two remaining issues.
  - `dicer` - No fix available
  Screenshot 2023-06-14 at 4 59 54 PM
  
  - `request` - [deprecated](https://www.npmjs.com/package/request). Replacing `request-promise` should resolve this. ([ticket](https://lite-farm.atlassian.net/browse/LF-3070))
  Screenshot 2023-06-14 at 5 01 04 PM

@kathyavini tagging you so that you could be aware of these changes (especially uninstalling xlsx). Please let me know if you see any problems!

Jira link: https://lite-farm.atlassian.net/browse/LF-3332

**Type of change**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

- [ ] Passes test case
- [ ] UI components visually reviewed on desktop view
- [ ] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

### CODE REVIEW

**Comment by kathyavini** [comment]
@SayakaOno thanks for the heads up! It looks like `xlsx` was refactored out of the sensor validation flow quite a while ago so it's all good 👍

**Comment by SayakaOno** [comment]
@kathyavini thank you so much for confirming!

File: packages/api/src/controllers/baseController.js [hunk 1]
```diff
@@ -25,13 +25,13 @@ import lodash from 'lodash';
 //
 //   async post(model, data, req, { trx, context = {} } = {}) {
 //     data = removeAdditionalProperties(model, data);
-//     return await model.query(trx).context({ user_id: req?.user?.user_id, ...context }).insert(data);
+//     return await model.query(trx).context({ user_id: req?.auth?.user_id, ...context }).insert(data);
```
Comment (iperdomo) [inline packages/api/src/controllers/baseController.js:28]: @SayakaOno can we please remove all this commented out code? I think is misleading

File: packages/api/tests/crop.test.js [hunk 2]
```diff
@@ -93,8 +93,8 @@ describe('Crop Tests', () => {
 
     // middleware = require('../src/middleware/acl/checkJwt');
     // middleware.mockImplementation((req, res, next) => {
-    //   req.user = {};
-    //   req.user.user_id = req.get('user_id');
+    //   req.auth = {};
+    //   req.auth.user_id = req.get('user_id');
```
Comment (iperdomo) [inline packages/api/tests/crop.test.js:97]: 🗑️ away

**Review by iperdomo — CHANGES_REQUESTED** [review]
Minor comments around modifying commented code. We have the git history. Let's remove the commented out code.

**Comment by Duncan-Brain** [comment]
Hallo!

Webapp
- Vite --host  --> we actually may expose this in our cypress tests github action we run pnpm dev on CI It would probably be good to update vite on another ticket.

Api
 - jwks-rsa --> can we just remove this? I do not see it used anywhere
 - override glob parent --> am I right in reading that once we update npm (node) this can be removed?
 - dicer --> I am not seeing the offending package on the newest lts for multer. Would adding a ticket to upgrade multer maybe fix it?

**Review by iperdomo — APPROVED** [review]

**Pull request merged**